File: app/src/main/java/com/botpa/turbophotos/main/DisplayLayoutManager.java
package com.botpa.turbophotos.main;

import android.content.Context;

import androidx.recyclerview.widget.LinearLayoutManager;

public class DisplayLayoutManager extends LinearLayoutManager {

    private boolean isScrollEnabled = true;


    //Layout manager
    public DisplayLayoutManager(Context context) {
        super(context);
    }

    //Scrolling
    public void setScrollEnabled(boolean flag) {
        this.isScrollEnabled = flag;
    }

    @Override
    public boolean canScrollHorizontally() {
        return isScrollEnabled && super.canScrollHorizontally();
    }

}

File: app/src/main/java/com/botpa/turbophotos/screens/display/DisplayLayoutManager.kt
package com.botpa.turbophotos.screens.display

import android.content.Context
import androidx.recyclerview.widget.LinearLayoutManager

class DisplayLayoutManager(context: Context) : LinearLayoutManager(context) {

    private var isScrollEnabled = true

    fun setScrollEnabled(flag: Boolean) {
        this.isScrollEnabled = flag
    }

    override fun canScrollHorizontally(): Boolean {
        return isScrollEnabled && super.canScrollHorizontally()
    }

}